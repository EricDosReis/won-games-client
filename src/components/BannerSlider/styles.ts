import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import * as BannerStyles from 'components/Banner/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    .slick-dots {
      display: flex !important;
      align-items: center;
      justify-content: center;
      list-style: none;
      margin-top: ${theme.spacings.small};

      li {
        background: ${theme.colors.white};
        border-radius: 100%;
        cursor: pointer;
        height: 1.2rem;
        width: 1.2rem;
        display: flex;
        align-items: center;
        justify-content: center;
        margin: 0 ${theme.spacings.xxsmall};

        &.slick-active {
          background: ${theme.colors.primary};
        }
      }

      button {
        cursor: pointer;
        height: 1.2rem;
        width: 1.2rem;
        opacity: 0;
      }
    }

    ${media.greaterThan('large')`
      ${BannerStyles.Wrapper} {
        margin: 0 auto;
        max-width: 104rem;
      }

      .slick-dots {
        flex-direction: column;
        height: 100%;
        position: absolute;
        top: 0;
        right: 0;
        margin: 0;

        li {
          margin: ${theme.spacings.xxsmall} 0;
        }
      }
    `}
  `}
`;
