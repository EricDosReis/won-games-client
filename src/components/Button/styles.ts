import styled, { css, DefaultTheme } from 'styled-components';
import { ButtonProps } from '.';

type WrapperProps = {
  hasIcon: boolean;
} & Pick<ButtonProps, 'size' | 'fullWidth'>;

const wrapperModifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
    height: 3rem;
  `,

  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
    height: 4rem;
    padding: ${theme.spacings.xxsmall} ${theme.spacings.medium};
  `,

  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    height: 5rem;
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xlarge};
  `,

  fullWidth: () => css`
    width: 100%;
  `,

  withIcon: (theme: DefaultTheme) => css`
    svg {
      width: 1.6rem;

      & + span {
        margin-left: ${theme.spacings.xxsmall};
      }
    }
  `,
};

export const Wrapper = styled.button<WrapperProps>`
  ${({ theme, size, fullWidth, hasIcon }) => css`
    background-image: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
    border: none;
    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    cursor: pointer;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    font-family: ${theme.font.family};
    padding: ${theme.spacings.xxsmall};
    text-decoration: none;

    &:hover,
    &:focus {
      background-image: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
    }

    ${!!size && wrapperModifiers[size](theme)};
    ${!!fullWidth && wrapperModifiers.fullWidth()}
    ${!!hasIcon && wrapperModifiers.withIcon(theme)}
  `}
`;
