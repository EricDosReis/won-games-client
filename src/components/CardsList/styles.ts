import styled, { css } from 'styled-components';

export const Card = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.lightGray};
    color: ${theme.colors.black};
    display: flex;
    align-items: center;
    padding: 1.3rem ${theme.spacings.xsmall};

    &:not(:last-child) {
      margin-bottom: ${theme.spacings.xsmall};
    }

    > span {
      margin-left: ${theme.spacings.xxsmall};
    }
  `}
`;
