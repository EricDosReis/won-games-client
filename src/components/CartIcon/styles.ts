import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    height: 2.4rem;
    width: 2.4rem;
    position: relative;
  `}
`;

export const Badge = styled.span`
  ${({ theme }) => css`
    background: ${theme.colors.secondary};
    color: ${theme.colors.white};
    display: flex;
    font-size: 1rem;
    font-weight: bold;
    align-items: center;
    justify-content: center;
    border-radius: 50%;
    height: 1.6rem;
    width: 1.6rem;
    position: absolute;
    top: -0.4rem;
    right: -0.4rem;
  `}
`;
