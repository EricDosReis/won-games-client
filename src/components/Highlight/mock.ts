export default {
  title: "Read Dead it's back!",
  subtitle: "Come see John's new adventures",
  buttonLabel: 'Buy now',
  buttonLink: '/red-dead-redemption-2',
  backgroundImage: '/img/highlight-bg.jpg',
};
