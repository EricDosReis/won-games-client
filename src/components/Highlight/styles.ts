import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import { HighlightProps } from '.';

type WrapperProps = Pick<HighlightProps, 'backgroundImage' | 'alignment'>;

const wrapperModifiers = {
  left: () => css`
    grid-template-areas: 'content float-image';
    grid-template-columns: 2fr 1.3fr;

    ${Content} {
      text-align: left;
    }

    ${FloatImage} {
      justify-self: end;
    }
  `,

  right: () => css`
    grid-template-areas: 'float-image content';
    grid-template-columns: 1.3fr 2fr;

    ${Content} {
      text-align: right;
    }
  `,
};

export const Wrapper = styled.div<WrapperProps>`
  ${({ backgroundImage, alignment }) => css`
    background-image: url(${backgroundImage});
    background-position: center center;
    background-size: cover;
    display: grid;
    height: 23rem;
    position: relative;

    &::after {
      content: '';
      background-color: rgba(0, 0, 0, 0.5);
      position: absolute;
      height: 100%;
      width: 100%;
    }

    ${media.greaterThan('medium')`
      height: 32rem;
    `}

    ${!!alignment && wrapperModifiers[alignment]()};
  `};
`;

export const Content = styled.div`
  ${({ theme }) => css`
    grid-area: content;
    padding: ${theme.spacings.xsmall};
    z-index: ${theme.layers.base};

    ${media.greaterThan('medium')`
      align-self: end;
      padding: ${theme.spacings.large};
    `};
  `};
`;

export const FloatImage = styled.img`
  ${({ theme }) => css`
    grid-area: float-image;
    align-self: end;
    max-height: 23rem;
    max-width: 100%;
    z-index: ${theme.layers.base};

    ${media.greaterThan('medium')`
      max-height: 32rem;
    `};
  `}
`;

export const Title = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `};
`;

export const Subtitle = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.light};
    margin-bottom: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `};
`;
