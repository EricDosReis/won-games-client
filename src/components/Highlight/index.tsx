import Button from 'components/Button';
import * as S from './styles';

export type HighlightProps = {
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
  backgroundImage: string;
  floatImage?: string;
  alignment?: 'left' | 'right';
};

const Highlight = ({
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  backgroundImage,
  floatImage,
  alignment = 'right',
}: HighlightProps) => (
  <S.Wrapper backgroundImage={backgroundImage} alignment={alignment}>
    {!!floatImage && <S.FloatImage src={floatImage} aria-label={title} />}

    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>

      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Content>
  </S.Wrapper>
);

export default Highlight;
