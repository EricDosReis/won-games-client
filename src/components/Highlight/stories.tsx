import { Story, Meta } from '@storybook/react/types-6-0';

import Highlight, { HighlightProps } from '.';

export default {
  title: 'Highlight',
  component: Highlight,
  args: {
    title: "Read Dead it's back",
    subtitle: "Come see John's new adventures",
    buttonLabel: 'Buy now',
    buttonLink: '/red-dead-redemption-2',
    backgroundImage: '/img/highlight-bg.jpg',
  },
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const Basic: Story<HighlightProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Highlight {...args} />
  </div>
);

export const withFloatImage: Story<HighlightProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Highlight floatImage="/img/highlight-float.png" {...args} />
  </div>
);
