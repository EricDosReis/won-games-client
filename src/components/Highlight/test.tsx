import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Highlight from '.';
import * as S from './styles';

const props = {
  title: 'Title',
  subtitle: 'Subtitle',
  buttonLabel: 'Buy now',
  buttonLink: '/red-dead-redemption-2',
  backgroundImage: '/img/red-dead-img.jpg',
};

describe('<Highlight />', () => {
  it('should render the highlight content', () => {
    const { container } = renderWithTheme(<Highlight {...props} />);

    expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument();
    expect(screen.getByText(/subtitle/i)).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with background image', () => {
    const { container } = renderWithTheme(<Highlight {...props} />);

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`,
    });
  });

  it('should render with float image', () => {
    renderWithTheme(<Highlight floatImage="/float-image.png" {...props} />);

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      '/float-image.png',
    );
  });

  it('should render right align by default', () => {
    const { container } = renderWithTheme(
      <Highlight floatImage="/float-image.png" {...props} />,
    );

    expect(container.firstChild).toHaveStyle({
      'grid-template-areas': "'float-image content'",
      'grid-template-columns': '1.3fr 2fr',
    });

    expect(container.firstChild).toHaveStyleRule('text-align', 'right', {
      modifier: `${S.Content}`,
    });
  });

  it('should render left align', () => {
    const { container } = renderWithTheme(
      <Highlight floatImage="/float-image.png" alignment="left" {...props} />,
    );

    expect(container.firstChild).toHaveStyle({
      'grid-template-areas': "'content float-image'",
      'grid-template-columns': '2fr 1.3fr',
    });

    expect(container.firstChild).toHaveStyleRule('text-align', 'left', {
      modifier: `${S.Content}`,
    });

    expect(container.firstChild).toHaveStyleRule('justify-self', 'end', {
      modifier: `${S.FloatImage}`,
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
