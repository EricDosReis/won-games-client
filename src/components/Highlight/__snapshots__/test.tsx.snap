// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render left align 1`] = `
.c7 {
  background-image: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: none;
  border-radius: 0.4rem;
  color: #FAFAFA;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.4rem;
  height: 4rem;
  padding: 0.8rem 3.2rem;
}

.c7:hover,
.c7:focus {
  background-image: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  display: grid;
  height: 23rem;
  position: relative;
  grid-template-areas: 'content float-image';
  grid-template-columns: 2fr 1.3fr;
}

.c0::after {
  content: '';
  background-color: rgba(0,0,0,0.5);
  position: absolute;
  height: 100%;
  width: 100%;
}

.c0 .c3 {
  text-align: left;
}

.c0 .c1 {
  justify-self: end;
}

.c4 {
  grid-area: content;
  padding: 1.6rem;
  z-index: 10;
}

.c2 {
  grid-area: float-image;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  max-height: 23rem;
  max-width: 100%;
  z-index: 10;
}

.c5 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 600;
}

.c6 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 300;
  margin-bottom: 2.4rem;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c4 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c2 {
    max-height: 32rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 1.8rem;
  }
}

<div
  class="c0"
>
  <img
    aria-label="Title"
    class="c1 c2"
    src="/float-image.png"
  />
  <div
    class="c3 c4"
  >
    <h3
      class="c5"
    >
      Title
    </h3>
    <p
      class="c6"
    >
      Subtitle
    </p>
    <a
      class="c7"
      href="/red-dead-redemption-2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</div>
`;

exports[`<Highlight /> should render the highlight content 1`] = `
.c5 {
  background-image: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: none;
  border-radius: 0.4rem;
  color: #FAFAFA;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.4rem;
  height: 4rem;
  padding: 0.8rem 3.2rem;
}

.c5:hover,
.c5:focus {
  background-image: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  display: grid;
  height: 23rem;
  position: relative;
  grid-template-areas: 'float-image content';
  grid-template-columns: 1.3fr 2fr;
}

.c0::after {
  content: '';
  background-color: rgba(0,0,0,0.5);
  position: absolute;
  height: 100%;
  width: 100%;
}

.c0 .c1 {
  text-align: right;
}

.c2 {
  grid-area: content;
  padding: 1.6rem;
  z-index: 10;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 600;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 300;
  margin-bottom: 2.4rem;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <h3
      class="c3"
    >
      Title
    </h3>
    <p
      class="c4"
    >
      Subtitle
    </p>
    <a
      class="c5"
      href="/red-dead-redemption-2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</div>
`;
